js/build.js

```diff
@@ -6,14 +6,19 @@ var Notifications = new Fliplet.Registry.get('fliplet-widget-notifications:1.0:c
 
   // This function will run for each instance found in the page
   Fliplet.Widget.instance('fliplet-widget-notifications-1-0-0', function (data) {
-    // Sample implementation to initialise the widget
+    // Sample implementation to initialize the widget
     notifications = new Notifications(data);
-    notifications.init();
-  });  
+  });
 
   Fliplet.Widget.register('Notifications', function () {
     return notifications;
   });
 
-  Fliplet.Hooks.run('notificationsReady');
+  Fliplet.Hooks.run('beforeNotificationsInit')
+    .then(notifications.init)
```

Comment:
@tonytlwu I don't get why you don't want to `init` when the hooks reject. Shouldn't you share the same instance?

Reply:
@squallstar This allows other components to stop the initialization if necessary. The instance can be accessed via `Fliplet.Widget.get('Notifications')` and initialized like https://github.com/Fliplet/fliplet-widget-notification-inbox/blob/7dfa8ec1e234d930aa90b07f88469e793b5c7e02/js/libs.js#L309-L316

Reply:
ah, I see. In that case it's all good with me! @tonytlwu